examples/c/src/L2_c_example.c

```diff
@@ -0,0 +1,105 @@
+#include <cuvs/core/c_api.h>
+#include <cuvs/distance/pairwise_distance.h>
+#include <cuvs/neighbors/cagra.h>
+
+#include <dlpack/dlpack.h>
+
+#include <cuda_runtime.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#define DIM 4
+#define N_ROWS 1
+
+float PointA[N_ROWS][DIM] = {1.0,2.0,3.0,4.0};
+float PointB[N_ROWS][DIM] = {2.0,3.0,4.0,5.0};
+
+void outputVector(float * Vec) {
+  printf("Vector is ");
+  for (int i = 0; i < DIM; ++i){
+    printf(" %f",Vec[i]);
+  }
+  printf("\n");
+}
+
+cuvsResources_t res;
+
+int initResources(void){
+  // Create a cuvsResources_t object
+  cuvsResourcesCreate(&res);
+  return 0;
+}
+
+int freeResources(void){
+  cuvsResourcesDestroy(res);
+  return 0;
+}
+
+/* Started by AICoder, pid:96cd0bcb6ad27b1149870b5f20ef16170f62a328 */
+void initializeTensor(float* x_d, int64_t x_shape[2], DLManagedTensor* x_tensor) {
+  x_tensor->dl_tensor.data = x_d;
+  x_tensor->dl_tensor.device.device_type = kDLCUDA;
+  x_tensor->dl_tensor.ndim = 2;
+  x_tensor->dl_tensor.dtype.code = kDLFloat;
+  x_tensor->dl_tensor.dtype.bits = 32;
+  x_tensor->dl_tensor.dtype.lanes = 1;
+  x_tensor->dl_tensor.shape = x_shape;
+  x_tensor->dl_tensor.strides = NULL;
+}
+/* Ended by AICoder, pid:96cd0bcb6ad27b1149870b5f20ef16170f62a328 */
+
+int calcL2Distance(int64_t n_cols,float x[], float y[], float *ret) {
```

Comment:
Can you provide some docs in this file, that briefly describe what the different steps are doing? Bonus points for function-level descriptions also (though not required).